File: packages/domain/notiz.ts
import { Aquarium } from "./aquarium";
import { Entity } from "./entity";

export interface Notiz extends Entity {
  id: string;
  text: string;
  datum: Date;
  tag: string;
  aquarium: Aquarium;
}

export interface NotizFormValues {
  id?: string;
  text?: string;
  datum: Date;
  tag?: string;
  aquarium: string;
}

export default Notiz;

File: Aqua.Domain/Notiz.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace com.marcelbenders.Aqua.Domain;

public class Notiz
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; }

    [BsonElement("userId")] public string UserId { get; set; }

    [BsonElement("text")] public string Text { get; set; }

    [BsonElement("datum")] public DateTimeOffset Datum { get; set; }

    [BsonElement("tag")] public string Tag { get; set; }

    [BsonElement("aquarium")] public Aquarium Aquarium { get; set; }

    [BsonIgnore] public string AquaType { get; private set; } = nameof(Notiz).ToLower();
}
